python/rikai/spark/utils.py

```diff
@@ -22,6 +22,7 @@
 def df_to_rikai(df: "pyspark.sql.DataFrame", uri: str):
     (
         df.write.format("rikai")
+        .mode("overwrite")
```

Comment:
why do we add `overwrite` everywhere?

Reply:
I think `overwrite` is the default behavior of the old save API. When we use the new API, the default behavior becomes `Error if exists`. And our test cases will reuse the same directory, so we need to declare it as `overwrite` explicitly to make the test case workable.

https://spark.apache.org/docs/latest/sql-data-sources-load-save-functions.html#save-modes

Reply:
Isnt the unit test create a new temporary directory each time? It supposes to be different / empty directory for each individual test.

Also this `df_to_rikai` is not in test package, so it is dangerous to give the default value to `overwrite`?

Reply:
I will remove the non-test overwrite code and to see what will happen

Reply:
In `RikaiRelationTest` and several other tests, data would be saved to the same directory many times.
https://github.com/eto-ai/rikai/blob/94cd2d16c43f55266c0c548cad4f5839bdf20e9b/src/test/scala/ai/eto/rikai/RikaiRelationTest.scala#L26

Reply:
I think I understand what happened, in the old writer, it will check if the directory is empty, if it's empty then it's safe to write. The new writer will check if it exists, if it exists, the write will fail no matter the path is empty or not.